src/core/backend/pearl/core.rs

```diff
@@ -0,0 +1,607 @@
+use crate::core::backend::core::*;
+use crate::core::backend;
+use crate::core::backend::pearl::{data::*, stuff::{LockGuard, Stuff}};
+use crate::core::configs::node::{NodeConfig, PearlConfig};
+use crate::core::data::{BobData, BobKey, VDiskId, VDiskMapper};
+use pearl::{Builder, Storage, ErrorKind};
+
+use futures03::{
+    future::err as err03,
+    task::{Spawn, SpawnExt},
+    FutureExt,
+};
+
+use std::{path::PathBuf, sync::Arc, thread};
+
+pub struct PearlBackend<TSpawner> {
+    vdisks: Arc<Vec<PearlVDisk<TSpawner>>>,
+    alien_dir: PearlVDisk<TSpawner>,
+}
+
+impl<TSpawner: Spawn + Clone + Send + 'static + Unpin + Sync> PearlBackend<TSpawner> {
+    pub fn new(mapper: VDiskMapper, config: &NodeConfig, spawner: TSpawner) -> Self {
+        debug!("initializing pearl backend");
+        let pearl_config = config.pearl.clone().unwrap();
+
+        let mut result = Vec::new();
+
+        //init pearl storages for each vdisk
+        for disk in mapper.local_disks().iter() {
+            let base_path = PathBuf::from(format!("{}/bob/", disk.path));
+            let mut vdisks: Vec<PearlVDisk<TSpawner>> = mapper
+                .get_vdisks_by_disk(&disk.name)
+                .iter()
+                .map(|vdisk_id| {
+                    let mut vdisk_path = base_path.clone();
+                    vdisk_path.push(format!("{}/", vdisk_id));
+                    
+                    PearlVDisk::new(
+                        &disk.path,
+                        &disk.name,
+                        vdisk_id.clone(),
+                        vdisk_path,
+                        pearl_config.clone(),
+                        spawner.clone(),
+                    )
+                })
+                .collect();
+            result.append(&mut vdisks);
+        }
+
+        //init alien storage
+        let path = format!(
+            "{}/alien/",
+            mapper
+                .get_disk_by_name(&pearl_config.alien_disk())
+                .unwrap()
+                .path
+        );
+
+        let alien_path = PathBuf::from(path.clone());
+        let alien_dir = PearlVDisk::new_alien(
+            &path,
+            &pearl_config.alien_disk(),
+            alien_path,
+            pearl_config.clone(),
+            spawner.clone(),
+        );
+
+        PearlBackend {
+            vdisks: Arc::new(result),
+            alien_dir: alien_dir,
+        }
+    }
+    
+    #[allow(dead_code)]
+    pub(crate) async fn test<TRet, F>(&self, disk_name: String, vdisk_id: VDiskId, f: F) -> BackendResult<TRet> 
+    where
+        F: Fn(&mut PearlSync) -> TRet + Send + Sync,
+    {
+        let vdisks = self.vdisks.clone();
+        let vdisk = vdisks.iter().find(|vd| vd.equal(&disk_name, &vdisk_id));
```

Comment:
this line (`vdisks.iter().find(|vd| vd.equal(&disk_name, &vdisk_id));`)  is repeating 4 times better to extract function